numba/np/arrayobj.py

```diff
@@ -3286,6 +3286,22 @@ def array_is_impl(a, b):
 
     return context.compile_internal(builder, array_is_impl, sig, args)
 
+
+@lower_builtin(operator.is_, types.Record, types.Record)
+def record_is(context, builder, sig, args):
+    aty, bty = sig.args
+    if aty != bty:
+        return cgutils.false_bit
+
+    def record_is_impl(a, b):
+        for field in aty.fields:
+            a_field = getattr(a, field)
+            b_field = getattr(b, field)
+            if a_field != b_field:
+                return cgutils.false_bit
```

Comment:
I think this won't work either:

```
E   numba.core.errors.TypingError: Failed in nopython mode pipeline (step: native lowering)
E   Failed in nopython mode pipeline (step: nopython frontend)
E   Unknown attribute 'fields' of type typeref[Record(a[type=int32;offset=0],b[type=int32;offset=4];8;False)]
E
```

Also, the `cgutils.false_bit` should be replaced by `return False` and it is missing a `return True` statement at the end.

Try to implement something similar to the array case.